src/script/index.js

```diff
@@ -24,10 +26,58 @@ for (var i = 0; i < paramPairsLength; i++) {
 delete i
 delete paramPair
 
+function add_instance(text, disabled, selected) {
+    const opt = document.createElement('option');
+    opt.innerText = text;
+    if (disabled) {
+        opt.setAttribute('disabled', true);
+    }
+    if (selected) {
+        opt.setAttribute('selected', true);
+    }
+    choose_instance.appendChild(opt);
+}
+
 if (prefillInstance != null) {
-    document.getElementById('instance').value = normalizeUrl(prefillInstance);
+    const url = normalizeUrl(prefillInstance);
+    instance.value = url;
+    add_instance(url.slice(8, url.length - 1), false, true);
 }
 
+choose_instance.addEventListener('focus', function (e) {
+    if (choose_instance.children.length < 3) {
+        if (choose_instance.children.length > 1) {
+            choose_instance.children[1].remove();
+        }
+        add_instance("... loading ...", true);
+        fetch('https://api.joinmastodon.org/servers')
+          .then(response => {
+              if (!response.ok) {
+                  throw new Error('Failure response from joinmastodon.');
+              }
+              return response.json();
+          })
+          .then(servers => {
+              choose_instance.children[1].remove();
+              const domains = servers.map(obj => obj.domain);
+              domains.sort();
+              for (const domain of domains) {
+                  add_instance(domain);
+              }
+          })
+          .catch(error => {
+              choose_instance.children[1].remove();
```

Comment:
> copy of the comment on an earlier line

If previously the instance was prefilled from the `localStorage`, this line will remove it, as it is technically at index 1.

Try

```suggestion
              choose_instance.children[choose_instance.children.length - 1].remove();
```

Reply:
Resolved by e27cb3f123dfb6ecc277cfb7320c97c2500a1ab6